homeassistant/components/vallox/__init__.py

```diff
@@ -132,6 +136,13 @@ def get_uuid(self) -> UUID | None:
             raise ValueError
         return uuid
 
+    def get_next_filter_change_date(self) -> date | None:
+        """Return the next filter change date."""
+        next_filter_change_date = calculate_next_filter_change_date(self.metric_cache)
+        if not isinstance(next_filter_change_date, date):
+            raise ValueError
```

Comment:
Hm, or should we just return `None` instead of throwing an error? 🤔 And possibly log something on debug level. Not sure if every single Vallox unit has this feature, and throwing an error will likely mean something is logged for each poll?

Reply:
```diff
diff --git homeassistant/components/vallox/__init__.py homeassistant/components/vallox/__init__.py
index 992dc27220..673d026e33 100644
--- homeassistant/components/vallox/__init__.py
+++ homeassistant/components/vallox/__init__.py
@@ -139,10 +139,14 @@ class ValloxState:
     def get_next_filter_change_date(self) -> date | None:
         """Return the next filter change date."""
         next_filter_change_date = calculate_next_filter_change_date(self.metric_cache)
+
         if not isinstance(next_filter_change_date, date):
-            raise ValueError(
-                f"Unexpected next filter change date type: {type(next_filter_change_date)}"
+            _LOGGER.debug(
+                "Unexpected next filter change date type: %s",
+                type(next_filter_change_date),
             )
+            return None
+
         return next_filter_change_date
 
 
diff --git tests/components/vallox/test_sensor.py tests/components/vallox/test_sensor.py
index 96eb88d692..4312a6ab91 100644
--- tests/components/vallox/test_sensor.py
+++ tests/components/vallox/test_sensor.py
@@ -91,7 +91,7 @@ async def test_remaining_time_for_filter_nothing_returned_from_vallox(
 
     # Assert
     sensor = hass.states.get("sensor.vallox_remaining_time_for_filter")
-    assert sensor is None
+    assert sensor.state == "unknown"
 
 
 @pytest.mark.parametrize(
```

Reply:
In my opinion, just `None` would be bad style. Don't know your integration in details, however, in general development, I would expect a `None` as a return value to be semantically be equivalent to "yeah, we tried, everything went normal, we just didn't find anything". Returning an error here could mean "ouh, sorry, with this input value I cannot proceed work". Just my 0.3 cents.

Reply:
Good points. I think it could make sense to do both then. The `vallox_websocket_api` returns `None` if the data was not available. In that case, I think it makes sense to return `None` here too. In any other case, throwing an error probably makes sense.